src/vs/workbench/contrib/debug/browser/disassemblyView.ts

```diff
@@ -0,0 +1,209 @@
+import { DISASSEMBLY_VIEW_ID } from 'vs/workbench/contrib/debug/common/debug';
+
+interface IDisassembledInstructionEntry {
+	allowBreakpoint: boolean;
+	isBreakpointSet: boolean;
+	instruction: DebugProtocol.DisassembledInstruction;
+}
+
+export class DisassemblyView extends EditorPane {
+
+	private static readonly NUM_INSTRUCTIONS_TO_LOAD = 50;
+
+	private _fontInfo: BareFontInfo;
+	private _disassembledInstructions: WorkbenchTable<IDisassembledInstructionEntry> | null;
+
+
+	constructor(
+		@ITelemetryService telemetryService: ITelemetryService,
+		@IThemeService themeService: IThemeService,
+		@IStorageService storageService: IStorageService,
+		@IConfigurationService configurationService: IConfigurationService,
+		@IInstantiationService private readonly _instantiationService: IInstantiationService
+	) {
+		super(DISASSEMBLY_VIEW_ID, telemetryService, themeService, storageService);
+
+		this._fontInfo = BareFontInfo.createFromRawSettings(configurationService.getValue('editor'), getZoomLevel(), getPixelRatio());
+		configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration('editor')) {
+				this._fontInfo = BareFontInfo.createFromRawSettings(configurationService.getValue('editor'), getZoomLevel(), getPixelRatio());
+				this._disassembledInstructions?.rerender();
+			}
+		});
+
+		this._disassembledInstructions = null;
+	}
+
+	protected createEditor(parent: HTMLElement): void {
+		const lineHeight = this._fontInfo.lineHeight;
+		const delegate = new class implements ITableVirtualDelegate<IDisassembledInstructionEntry>{
+			headerRowHeight: number = 0; // No header
+			getHeight(row: IDisassembledInstructionEntry): number {
+				return lineHeight;
+			}
+		};
+
+		this._disassembledInstructions = this._register(this._instantiationService.createInstance(WorkbenchTable,
+			'DisassemblyView', parent, delegate,
+			[
+				{
+					label: '',
+					tooltip: '',
+					weight: 0,
+					minimumWidth: 40,
+					maximumWidth: 40,
+					templateId: BreakpointRenderer.TEMPLATE_ID,
+					project(row: IDisassembledInstructionEntry): IDisassembledInstructionEntry { return row; }
+				},
+				{
+					label: 'instructions',
+					tooltip: '',
+					weight: 0.3,
+					templateId: InstructionRenderer.TEMPLATE_ID,
+					project(row: IDisassembledInstructionEntry): IDisassembledInstructionEntry { return row; }
+				},
+			],
+			[
+				this._instantiationService.createInstance(BreakpointRenderer),
+				this._instantiationService.createInstance(InstructionRenderer),
+			],
+			{
+				identityProvider: { getId: (e: IDisassembledInstructionEntry) => e.instruction.address },
+				horizontalScrolling: false,
+				overrideStyles: {
+					listBackground: editorBackground
+				},
+				multipleSelectionSupport: false,
+				setRowLineHeight: false,
+				openOnSingleClick: false,
+			}
+		)) as WorkbenchTable<IDisassembledInstructionEntry>;
+
+		this.loadDisassembledInstructions(0, DisassemblyView.NUM_INSTRUCTIONS_TO_LOAD * 2, '0x00005000');
```

Comment:
I find it weird that you called the constant NUM_INSTRUCTIONS_TO_LOAD but then you load twice as much.
I think the constant should be larger and then you would either use it or it divided by 2, if that makes sense?

Reply:
I consider it as number of instructions to load each time. Since the current instruction is revealed in the middle of the viewport, the first load actually loads "twice": "first time" before the current instruction, and "second time" include and after the current instruction.

Reply:
Okey. I was just sharing that I found it a bit confusing :)